src/__internal__/focus-trap/focus-trap-utils.js

```diff
@@ -40,27 +40,67 @@ const isRadio = (element) => {
   );
 };
 
-const nextNonRadioElementIndex = (element, focusableElements) => {
+const getRadioElementToFocus = (groupName, shifKey) => {
+  const buttonsInGroup = document.querySelectorAll(
+    `input[type="radio"][name="${groupName}"]`
+  );
+  const selectedButton = [...buttonsInGroup].find((button) => button.checked);
+
+  if (selectedButton) {
+    return selectedButton;
+  }
+  return buttonsInGroup[shifKey ? buttonsInGroup.length - 1 : 0];
+};
+
+const getNextElement = (element, focusableElements, shiftKey) => {
   const currentIndex = focusableElements.indexOf(element);
-  let nextIndex = currentIndex - 1;
+  const increment = shiftKey ? -1 : 1;
+  let nextIndex = currentIndex;
+  let foundElement;
+
+  while (!foundElement) {
+    nextIndex += increment;
+    if (nextIndex < 0) {
+      nextIndex += focusableElements.length;
+    }
+    if (nextIndex >= focusableElements.length) {
+      nextIndex -= focusableElements.length;
+    }
+
+    const nextElement = focusableElements[nextIndex];
+
+    if (nextElement === element) {
```

Comment:
hi Ed, having looked at this I don't think there's any real benefit to adding this.

If focusableElements.length is 1, then the simple nextIndex calculations above will ensure that nextElement is the same as element, so it'll full into the guard in the while loop. And while that's is doing a little bit more, it's only adding and subtracting one, and indexing into a 1-element array, and that inside a loop that will only run once in this case, so I honestly don't see any performance impact. This strikes me as a micro-optimisation with no real benefit and which will make the code more confusing. We can always revisit in the unlikely event that focus traps with a single focusable element (which seems unlikely in practice anyway) are found to cause a performance issue.

Reply:
I'm happy to leave it as is as it's likely to be a rare use case and there is never a case that I can think of when we would have multiple traps running at once (even if there is there would need to be a lot) so the cost isn't too high when the above scenario happens. I commented the above as we actually provide Modals that would match 1 focusable element, for example https://carbon.sage.com/?path=/story/confirm--single-action and with current implementation it would be doing checks and assignments it doesn't need to do.